insights/client/connection.py

```diff
@@ -663,25 +663,30 @@ def _legacy_api_registration_check(self):
         #       False for unregistered
         #       None for system 404
         if res.status_code != 200:
-            # Use handle_fail_rcs to log the error response clearly to the user
             self.handle_fail_rcs(res)
-        try:
-            # check the 'unregistered_at' key of the response
-            unreg_status = json.loads(res.content).get('unregistered_at', 'undefined')
-            # set the global account number
-            self.config.account_number = json.loads(res.content).get('account_number', 'undefined')
-        except ValueError:
-            # bad response, no json object
+        if res.status_code not in (200, 404):
+            # Network error returns False
             return False
-        if unreg_status == 'undefined':
-            # key not found, machine not yet registered
-            return None
-        elif unreg_status is None:
-            # unregistered_at = null, means this machine IS registered
-            return True
         else:
-            # machine has been unregistered, this is a timestamp
-            return unreg_status
+            try:
+                # check the 'unregistered_at' key of the response
+                unreg_status = json.loads(res.content).get('unregistered_at', 'undefined')
+                # set the global account number
+                self.config.account_number = json.loads(res.content).get('account_number', 'undefined')
+            except ValueError:
+                # bad response, no json object
+                return False
+            if unreg_status == 'undefined':
+                # key not found, machine not yet registered
+                return None
+            elif unreg_status is None:
+                # unregistered_at = null, means this machine IS registered
+                return True
+            else:
+                # machine has been unregistered, this is a timestamp
+                # This is done for legacy servers that responded with the timestamp of disconnection
+                # TODO: consider to remove this condition
```

Comment:
I couldn't reproduce this response status code 200 and the timestamp. But it seemed to be a correct response to the legacy API. So that's why I marked to remove. Yet, it needs further investigation.

I have to disagree with the fact that the return value doesn't affect anything. The return value affects the execution. It is evaluated by [_legacy_registration_check()](https://github.com/ahitacat/insights-core/blob/d98f9cd2e4d62b4b9787ff06023526b58900863e/insights/client/support.py#L24) and determines if the system is registered via API or not. This value is used by the [registration_check](https://github.com/ahitacat/insights-core/blob/d98f9cd2e4d62b4b9787ff06023526b58900863e/insights/client/support.py#L59).

The _legacy_api_registration_check performs a call to the API, which needs the machine-id. This value is obtained by generating_machine_id and logging the process. Note that this method only gets the machine ID and doesn't generate a new one (yes, technical debt). Although it is not the best manner to code this, it is minimal a change to do the fix with all the technical debt behind the code.

About the PR you mentioned, it added the removal of the machine-id file when it detected it was unregistered. It uses the code that was already there. The return value is operated the same way it was in [_legacy_registration_check()](https://github.com/ahitacat/insights-core/blob/d98f9cd2e4d62b4b9787ff06023526b58900863e/insights/client/support.py#L24) and it adds the removing bits.